client/blb/client_test.go

```diff
@@ -257,6 +304,376 @@ func TestWriteReadTracedExactlyOneTract(t *testing.T) {
 	trace.checkLength(t, 7)
 }
 
+// setupBackupRequestFunc overrides spawning logic for backup reads, it will spawn nReaders
+// goroutines that will block on respective delayChans. To release a reader one can write to the
+// delayChan.
+func setupBackupRequestFunc(cli *Client, nReaders int) []chan time.Time {
+	delayChans := make([]chan time.Time, nReaders)
+	for i := 0; i < nReaders; i++ {
+		delayChans[i] = make(chan time.Time)
+	}
+
+	resultCh := make(chan tractResultRepl)
+	cli.backupRequestFunc = func(
+		ctx context.Context,
+		cli *Client,
+		reqID string,
+		tract *core.TractInfo,
+		thisB []byte,
+		thisOffset int64,
+		nReaders int,
+		order []int) <-chan tractResultRepl {
+
+		for i, ch := range delayChans {
+			go func(i int, delayCh <-chan time.Time) {
+				cli.readOneTractBackupReplicated(ctx, reqID, tract,
+					thisB, thisOffset, delayChans[i], resultCh, order[i])
+			}(i, ch)
+		}
+		return resultCh
+	}
+	return delayChans
+}
+
+// setupBackupClient initializes a client that has backup requests enabled.
+// it also overrides synchronization hooks on the client that allow for
+// contol of read behavior operation ordering.
+func (cli *Client) setupBackupClient(maxNumBackups int, overrideDelay bool) (<-chan bool, <-chan bool, <-chan bool) {
+	behavior := BackupReadBehavior{
+		Enabled:       true,
+		MaxNumBackups: maxNumBackups - 1,
```

Comment:
Having two things both named "max num backups" but with slightly different semantics (off by one) is really confusing. Pick one semantics and use it consistently, or else rename one of these.